rules/directive-restrict.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+module.exports = function(context) {
+    var utils = require('./utils/utils');
+
+    var options = context.options[0] || {};
+    var restrictOpt = options.restrict || 'AE';
+    var explicitRestrict = options.explicit === 'always';
+    var restrictChars = [];
+
+    restrictOpt.split('').forEach(function(char) {
+        if ('ACEM'.indexOf(char) === -1) {
```

Comment:
I figured users could potentially misconfigure the acceptible restrict as for example `AEA`. I don't know how to check this using just a pattern. Should I just assume users won't do this?